src/components/Table/useColumns.ts

```diff
@@ -0,0 +1,21 @@
+import { Dispatch, SetStateAction, useMemo, useState } from "react";
+import { GridColumn, Kinded } from "./GridTable";
+
+export interface Columns<R extends Kinded, S = {}> {
```

Comment:
**nit/suggestion:** I feel like `Columns` is too generic, but I can't think of a better name. Maybe something like: `GridColumnState`?